File: SproInstall/SocketPro/samples/performance/csharp/PerformanceStudyDotNetRemoting/DefMyCallsInterface/defmycallsinterface.cs
//#define USE_SQLCLIENT

#if USE_SQLCLIENT
using System.Data.SqlClient;
#else
using System.Data.OleDb;
#endif

using System;
using System.Data;


namespace DefMyCallsInterface
{
    public interface IMyCalls
    {
        DataTable OpenRowset(string strSQL);
        string MyEcho(string strInput);
    }

    public class CMyCallsImpl : MarshalByRefObject, IMyCalls
    {
#if USE_SQLCLIENT
        SqlConnection conn = new SqlConnection("server=localhost\\sqlexpress;Integrated Security=SSPI;database=northwind");
#else
        OleDbConnection conn = new OleDbConnection("Provider=Microsoft.Jet.OLEDB.4.0;Data Source=C:\\Program Files\\udaparts\\SocketPro\\bin\\nwind3.mdb");
#endif
#region IMyCalls Members
        public System.Data.DataTable OpenRowset(string strSQL)
        {
            DataTable dt = null;
            if(conn.State != ConnectionState.Open)
                conn.Open();
#if USE_SQLCLIENT
            SqlCommand cmd = new SqlCommand(strSQL, conn);
            SqlDataAdapter adapter = new SqlDataAdapter(cmd);
#else
            OleDbCommand cmd = new OleDbCommand(strSQL, conn);
            OleDbDataAdapter adapter = new OleDbDataAdapter(cmd);             
#endif
            dt = new DataTable("MyDataSet");
            adapter.Fill(dt);
//            ds.SchemaSerializationMode = SchemaSerializationMode.ExcludeSchema;
            dt.RemotingFormat = SerializationFormat.Binary;
            return dt;
        }
        public string MyEcho(string strInput)
        {
            return strInput;
        }
#endregion
    }
}

File: SproInstall/SocketPro/samples/performance/vbnet/PerformanceStudyDotNetRemoting/DefMyCallsInterface/defmycallsinterface.vb
#Const USE_SQLCLIENT = False

#If USE_SQLCLIENT Then

Imports Microsoft.VisualBasic
Imports System.Data.SqlClient
#Else
Imports System.Data.OleDb
#End If

Imports System
Imports System.Data


Namespace DefMyCallsInterface
	Public Interface IMyCalls
        Function OpenRowset(ByVal strSQL As String) As DataTable
		Function MyEcho(ByVal strInput As String) As String
	End Interface

	Public Class CMyCallsImpl
		Inherits MarshalByRefObject
        Implements IMyCalls

#If USE_SQLCLIENT Then
        Dim conn As SqlConnection = New SqlConnection("server=localhost\sqlexpress;Integrated Security=SSPI;database=northwind")
#Else
        Dim conn As OleDbConnection = New OleDbConnection("Provider=Microsoft.Jet.OLEDB.4.0;Data Source=C:\Program Files\udaparts\SocketPro\bin\nwind3.mdb")
#End If

#Region "IMyCalls Members"
        Public Function OpenRowset(ByVal strSQL As String) As System.Data.DataTable Implements IMyCalls.OpenRowset
            Dim dt As DataTable = Nothing
            If conn.State <> ConnectionState.Open Then
                conn.Open()
            End If
#If USE_SQLCLIENT Then
            Dim cmd As SqlCommand = New SqlCommand(strSQL, conn)
            Dim adapter As SqlDataAdapter = New SqlDataAdapter(cmd)
#Else
			Dim cmd As OleDbCommand = New OleDbCommand(strSQL, conn)
			Dim adapter As OleDbDataAdapter = New OleDbDataAdapter(cmd)
#End If
            dt = New DataTable("MyDataTable")
            adapter.Fill(dt)
            'dt.SchemaSerializationMode = SchemaSerializationMode.ExcludeSchema;
            dt.RemotingFormat = SerializationFormat.Binary
            Return dt
        End Function
		Public Function MyEcho(ByVal strInput As String) As String Implements IMyCalls.MyEcho
			Return strInput
		End Function
#End Region

	End Class
End Namespace
